### PR TITLE
Make NLE only run bzip2 when necessary

### PR DESCRIPTION
**Pull request by heiner**
Also fix misc test again and make installation a bit quieter.

### CODE REVIEW

File: src/nle.c [hunk 1]
```diff
@@ -184,21 +185,23 @@ mainloop(fcontext_transfer_t ctx_transfer)
 }
 
 boolean
-write_data(void *buf, int length)
+write_ttyrec_data(void *buf, int length)
 {
     nle_ctx_t *nle = current_nle_ctx;
+    if (nle->ttyrec) {
```
Comment (condnsdmatters) [inline src/nle.c:191]: do we need this check here given we have it below in nle_fflush? I understand trying to be defensive, but maybe we should pick one location/level of abstraction at which to put the if...

**Review by condnsdmatters — APPROVED** [review]
Only one comment, surrounding a potential extra if statement we might not need

File: src/nle.c (hunk 1 above)
Comment (heiner) [reply src/nle.c:191]: Wait this is a weird place to ask this question -- if `nle->ttyrec` is null the `BZ2_bzWrite` call below will surely fail as `nle->ttyrec_bz2` won't be a valid bz2 file?

Comment (heiner) [reply src/nle.c:191]: Ah sorry, now I see what you mean: This function `write_ttyrec_data` never gets called with a NULL `ttyrec`. Fair point!

Comment (heiner) [reply src/nle.c]: Removed the `if`, sorry for being slow in understanding.

**Pull request merged**